app/assets/javascripts/bind_event_to_preview_page.js

```diff
@@ -1,17 +1,37 @@
 $(document).ready(function () {
   var event_date = $('.parallax-container').data('countdown'),
+      end_date = $('.end_date').data('countdown'),
+      start_time = $('.start_time').data('countdown'),
+      end_time = $('.end_time').data('countdown'),
       prev_color = '',
       color = '',
       map;
 
-  if (event_date) { countdown(convertDate(event_date)); }
+  if (event_date) { countdown(convertDate(event_date, start_time), end_date, end_time); }
 
   $('.preview').click(function () {
     var map_val = $('#event_map_url').val(),
         start_date = $('#event_start_date').val(),
-        end_date = $('#event_end_date').val();
+        end_date = $('#event_end_date').val(),
+        start_time = $('#event_start_time').val(),
+        end_time = $('#event_end_time').val(),
+        frequency = $('#event_recurring_event_attributes_frequency').val(),
+        week = $('#event_recurring_event_attributes_week').val(),
+        day = $('#event_recurring_event_attributes_day').val();
+
+    var rec_text;
+    if (frequency === 'Daily'){
+      rec_text = 'Every Day from ';
+    } else if (frequency === 'Weekly'){
+      rec_text = 'Every ' + day + ' from ';
+    } else if (frequency === 'Monthly'){
+      rec_text = 'Every ' + week + ' ' + day + ' of the Month, from ';
+    } else {
+      rec_text = '';
```

Comment:
Can make use of `Switch-Case` here for more optimization, as well as readability.